packages/cozy-client/src/helpers/settings.js

```diff
@@ -0,0 +1,188 @@
+ */
+export const saveAfterFetchSetting = async (
+  client,
+  slug,
+  key,
+  valueOrSetter
+) => {
+  const query = getQuery(slug)
+
+  const currentSettingsResult = await client.fetchQueryAndGetFromState({
+    definition: query.definition,
+    options: query.options
+  })
+
+  const currentSettings = normalizeSettings(currentSettingsResult.data)
+
+  let value = undefined
+  if (typeof valueOrSetter === 'function') {
+    value = valueOrSetter(currentSettings[key])
+  } else {
+    value = valueOrSetter
+  }
+
+  const newSettings = editSettings(slug, currentSettings, key, value)
+
+  return await client.save(newSettings)
+}
+
+/**
+ * Convert a result from a `client.query()` or a `useQuery()` that can be
+ * an object or an array of objects into a single object
+ *
+ * @param {Array | Object} data - Result from a client.query or a useQuery
+ * @returns {Object} A single object containing the setting data
+ */
+export const normalizeSettings = data => {
+  const settingsData = Array.isArray(data) ? data[0] : data
+
+  return settingsData || {}
+}
+
+/**
+ * Edit the given settings by injecting `value` into the `key` entry
+ * This methods takes care of the kind of settings that is edited as there are
+ * some exceptions in settings formats (i.e. `io.cozy.settings.instance`)
+ *
+ * @param {string} slug - the cozy-app's slug containing the setting (can be 'instance' for global settings)
+ * @param {Object} currentSettings - the Setting object (ideally from a `client.query()` or a `useQuery()` and normalized using `normalizeSettings`)
+ * @param {string} key - The name of the setting to edit
+ * @param {any} value - The new value for the setting
+ * @returns {Object} a new Setting object containing the new value
+ */
+export const editSettings = (slug, currentSettings, key, value) => {
+  const type = getDoctype(slug)
+
+  const newSettings =
+    slug === 'instance'
+      ? mergeInstance(currentSettings, key, value)
+      : mergeSettings(currentSettings, type, key, value)
+
+  return newSettings
+}
+
+const mergeInstance = (currentSettings, key, value) => {
+  return {
+    _id: currentSettings._id,
+    _type: currentSettings._type,
+    _rev: currentSettings.meta.rev,
+    ...currentSettings,
+    attributes: {
+      ...currentSettings.attributes,
+      [key]: value
+    },
+    [key]: value
+  }
+}
+
+const mergeSettings = (currentSettings, type, key, value) => {
+  return {
+    _type: type,
+    ...currentSettings,
+    [key]: value
+  }
+}
+
+/**
+ * Create a Query that can be used to fetch the cozy-app settings for the given slug
+ *
+ * @param {string} slug - the cozy-app's slug containing the setting (can be 'instance' for global settings)
+ * @returns {import('../types').Query} - the Query that can be used to fetch the cozy-app settings
+ */
+export const getQuery = slug => {
+  if (slug === 'instance') {
+    return getNestedSettings(slug)
+  }
+
+  if (slug === 'passwords') {
+    return getNestedSettings('bitwarden')
+  }
```

Comment:
I feel like 'instance' and 'passwords' should be explicitly mentioned in the doc + jsdoc, as it is not obvious. `instance` is mentioned as an example, but not `passwords`, and the latter is particularly not intuitive

Reply:
Documented [here](https://github.com/cozy/cozy-client/pull/1460/commits/c53f32c5da6d49f3de2d1a644f34505710c27d85)